README.md

```diff
@@ -173,6 +173,20 @@ ignores: [
 
 In this way, you can control which files the UI5 linter should process and which it should ignore.
 
+## Internals
+
+UI5 linter makes use of the [TypeScript compiler](https://github.com/microsoft/TypeScript/) to parse and analyze the source code (both JavaScript as well as TypesScript) of a UI5 project. This allows a decent level of accuracy and performance.
+
+For this UI5 linter provides the TypeScript compiler with the SAPUI5 [type definitions](https://sap.github.io/ui5-typescript/), the compiler can then infer the usage of UI5 classes and modules in the code. UI5 linter only needs to ask the compiler whether a given function or property is deprecated and the compiler will provide it with the full deprecation text.
+
+In some cases UI5 linter will transpile some of the source files before analyzing them. For example the `sap.ui.require` syntax is transpiled to `import` statements (basically ESM) so that the TypeScript compiler can understand them.
```

Comment:
```suggestion
In some cases, UI5 linter will transpile source files before analyzing them. For example, the `sap.ui.require` syntax is transpiled to `import` statements (basically ESM), so that the TypeScript compiler can understand them.
```